src/Microsoft.DotNet.Configurer/NuGetCacheSentinel.cs

```diff
@@ -30,7 +30,7 @@ private string NuGetCachePath
             }
         }
 
-        private string SentinelPath => Path.Combine(NuGetCachePath, SENTINEL);
+        private string _sentinelPath => Path.Combine(NuGetCachePath, SENTINEL);
```

Comment:
That should be ok. You are going to have to move the using clause for the dotnetsentinel out and pass it in. But I still think that is better than leaking the path information. See how you had to plumb it through so many objects that have no business knowing that information, specially when you have the sentinel available much closer.